### PR TITLE
idc.py: renamed 'SizeOf' to 'sizeof' to be compatible with IDC

### PR DESCRIPTION
**Pull request by 0xeb**
Aliased and kept the original 'SizeOf'

### CODE REVIEW

**Comment by assarbad** [comment]
This function appears to have an issue, which I have already reported to the Hex-Rays support.

When passing a name into it, it complains about expecting a string. @skochinsky explained that it actually expects bytes instead.

I was advised to use `tinfo_t` instead and came up with the following based on Igor's guidance:

```
def sizeof(typestr: str) -> int:
    ti = ida_typeinf.tinfo_t()
    if ti.get_named_type(None, typestr):
        return ti.get_size()
    return -1
```

**Comment by 0xeb** [comment]
Yes and no.

I also had a discussion w/ support@ and the team.

Basically, it has to mimic the IDC version which supports either a string or a type string:
 
/// Calculate the size of a type
///      type - type to calculate the size of
///             can be specified as a typeinfo object (e.g. the result of get_tinfo())
///             or a string with C declaration (e.g. "int")
/// returns: size of the type or -1 if error

long sizeof(typeinfo type);


I will fix it to support both cases.

**Comment by assarbad** [comment]
Makes sense. I just didn't know how close you were still working with folks from Hex-Rays and so wanted to bring this to your attention as well. Thanks for your efforts.

**Comment by 0xeb** [comment]
I appreciate more eyes looking and taking care of this project. I stay in touch via support@ like most clients.

File: python/idc.py [hunk 1]
```diff
@@ -4978,12 +4978,22 @@ def get_type(ea):
     """
     return ida_typeinf.idc_get_type(ea)
 
-def sizeof(typestr):
+def sizeof(tp):
     """
     Returns the size of the type. It is equivalent to IDC's sizeof().
+	You may pass:
```
Comment (assarbad) [inline python/idc.py:4984]: Seems you are mixing tabs and space, but it's just cosmetics, of course.

**Comment by 0xeb** [comment]
Oh no, let me fix it.

**Comment by 0xeb** [comment]
did you join their team? (I don't know)

**Comment by assarbad** [comment]
> did you join their team? (I don't know)

If you mean if I work for Hex-Rays, nope I don't. Also merely a customer and RCE enthusiast. But I attended a seminar you gave about IDA plugin writing several years ago.

**Comment by 0xeb** [comment]
@assarbad , imagine that Oliver. After waiting for this PR, 3 weeks, I finally ping Hex-Rays and they told me they fixed it on their way.
Way to go for 'open source' and 'collaboration'.

Makes you wonder if it is worthwhile the efforts to engage in this project at all.

**Pull request closed without merging**